lib/ansible/parsing/vault/__init__.py

```diff
@@ -349,17 +349,19 @@ def script_is_client(filename):
 
 
 def get_file_vault_secret(filename=None, vault_id=None, encoding=None, loader=None):
-    this_path = os.path.realpath(os.path.expanduser(filename))
 
+    # while files from --vault-password-file are unfracked, other sources are not
+    this_path = unfrackpath(filename, follow=False)
```

Comment:
It's already been through unfrackpath once at this point in our code, but I guess this would be helpful for other callers.